### ISSUE TITLE
Cannot use enum from project A from project B even if the function used in B is imported from A

### ISSUE THREAD

**Issue by kenotron** [Microsoft/TypeScript#5382]
Error message says: 

```
error TS2345: Argument of type 'Foo' is not assignable to parameter of type 'Foo'.
```
Given the exact same exported type, Typescript isn't able to deal with enums from multiple source locations. Here's an example with a Core project and a Bar project:

```
Core/
  foo.ts
  convertFooToString.ts
  dist/
    foo.js
    foo.d.ts
    convertFooToString.ts
    convertFooToString.d.ts

Bar/
  SomeComponent.tsx
  dist/
    SomeComponent.js
    SomeComponent.d.ts
```

SomeComponent.tsx:
```js
// some other imports
import {convertFooToString}
export default class SomeComponent extends React.Component<any, any> {
  render() {
    // Note, this.props.foo IS a Foo type
    return (<div>{convertFooToString(this.props.foo)}</div>);
  }
}
```

foo.ts:
```js
enum Foo {
  Yay = 0,
  Nay = 1
};

export default Foo;
```

convertFooToString:
```js
export function convertFooToString(foo: Foo): string {
    switch (foo) {
        case Foo.Yay:
            return "YAYAYAYAYA";
        default:
            return "NANANANANA";
    }
}
```

**Comment by mhegazy** [comment]
it is not clear what is going on from your description, i do not know where the error is reported, and where hte import is coming from etc.. but this looks like a duplicate of #4800..

**Comment by xogeny** [comment]
This issue was long ago marked as a duplicate.  Shouldn't it be closed (especially since the issue it is supposedly a duplicate of, #4800, is actually closed)?
